ls1intum/Artemis
# `General`: Fix navigation highlighting for 'Assessment' link

### Checklist
#### General

#### Client

### Motivation and Context
This PR fixes a bug in the course management tab bar. Before the fix, it was possible to have two links _(Exams & Assessment)_ highlighted at the same time when navigating to an exam's grading key.

### Description
The shouldHighlightAssessmentLink() function now excludes all links containing "exams".

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Exam with grading key

1. Navigate to the exam grading key
2. Note that only the exam tab is highlighted
3. Make sure that the assessment links still work as expected

Additional:
- Navigate around and verify that the same bug does not appear in other places.

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [course-management-tab-bar.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5300/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts.html) | 96.77% | ✅ |

### Screenshots
Before the bugfix, the _exam_ and _assessment dashboard_ link were highlighted on the exam grading key page.
![Bildschirmfoto 2023-08-11 um 07 22 24](https://github.com/ls1intum/Artemis/assets/47261058/145154ce-2632-4c80-85f4-d9a1c04b01ae)
